share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,809 @@
+
+3. Create a wake database: `wake --init .`.After this you will see a `wake.db` file created inside the folder.
+
+4. Create a file with `.wake` extension and copy the functions mentioned in this tutorial and run using `wake` command in the terminal as mentioned in Step 5.  
+
+5. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`. Ex: `wake -x 'double 2'`.
+   
+   `v` stands for verbosity and `x` stands for execute the expression.
+
+6. To know about various wake options run `wake --help`. 
+
+7. For more information on wake library click [here](https://sifive.github.io/wake/).
+
+### wake `Hello World`
+* If you want to display a string you can do that using the below function by writing the string to be displayed inside "" and call the function `abc`.
+
+Copy the below function into a `.wake` file you created in step 4
+```
+def abc = "Hello World"
+```
+Then execute the following command 
+```
+wake -x 'abc'
+```
+This will call the function `abc` and print `Hello World`.
+
+* You can also do this without a `.wake` file by doing it all in one command: `wake -x '"Hello World"'`
+
+### Defining Functions
+A function should be defined using the keyword `def`. Simple function definition takes the form
+`def <function name> <parameter list> = <expression>`
+
+We cannot specify explicitly the return type and type of the parameters. Usually when the function body is of multi-line the return value is specified by the last line in the function.
+
+Note: The only thing you can define functions are using `def` keyword. In wake we deal everything with functions.
+
+#### Few operators that work on Integer data type
+* `+` Addition Operator
+* `-` Subtraction Operator
+* `*` Multiplication Operator
+* `/` Division Operator
+* `^` Exponent Operator
+* `%` Modulus Operator
+* `>>` Right shift Operator
+* `<<` Left shift Operator
+
+#### Few operators that work on Double data type
+* `+.` Addition Operator
+* `-.` Subtraction Operator
+* `*.` Multiplication Operator
+* `/.` Division Operator
+* `^.` Exponent Operator
+
+* Below function  double takes an argment x and gives the output by multiplying the x by 2
+```
+def double x = 2*x
+```
+
+* Below function triple takes an argument x and gives the output by multiplying the x by 3
+```
+def triple x = 3*x
+```
+
+* Function inc increments the values by 1 and returns the value
+```
+def inc x = x+1
+```
+
+* Function adda takes an argument of type string and concatenates it with b and gives the output. `{ }` is the string interpolation operator
+```
+def adda s = "{s}b"
+```
+
+* Function times4 takes an argument x and applies the function double twice on x
+```
+def times4 x = double (double x)
+```
+
+* Adds 2 integer values and gives out the result
+```
+def sumadd x y = x+y 
+```
+
+* power 2 3 = 8
+```
+def power x n = x^n
+```
+
+#### lambda operator (\)
+The lambda operator can be used to define paramaters of a function. 
```

Comment:
```suggestion
The lambda operator `\` is an alternative way to define the parameters of a function. 
```

Reply:
High level `wake` question... we dont seem to use this in any of the examples. Can we just discourage its use? Why do we need both?

Reply:
ok actually you do show it later, but not in this way. Could you give an example of the form where you use it in an anonymous function  right away here? Or save the introduction of the lambda operator til you get to the map / higher order functions explanation